### PR TITLE
Kieran Blazier PA2

### PR DESCRIPTION
**Pull request by yes91**
Please check "Assemble all files in current directory" before testing parts 2 & 3. They require globally defined labels in string.asm.

### CODE REVIEW

**Comment by JShrein** [comment]
**\* I admire that you decided to finish the work, but your later commits were a bit after the fact.  If you can convince Dr. Strange that the late portion should be accepted then I am happy to regrade it.  If you see any mistakes or have any questions then feel free to ask.  Also, my comments may be difficult to read because github is killing my formatting ***

PART 1:
Total: 30/30

Compiles and executes correctly
No critical mistakes
All questions answered

PART 2:
Total: 20/25

Your atoi() function works correctly
Your selection_sort() works correctly
Your itoa() does not work entirely correctly.  The problem occurs when you're popping your number off the stack and putting it back in buffer.  Its actually correct in the stack, but when you put it into the buffer, you're putting it in backwards and after the first byte each following byte is incorrectly offset by 4x greater than it should be.  For example, let's assume your data buffer is just null space (which it isn't) 
This is what it looks like (arranged like it is in MARS it reads from low to high order):
0x100100cc     0x100100d0     0x100100d4     0x100100d8    <--- this is word address
\0 \0 \0 \0           \0 \0 \0 \0          \0 \0 \0 \0            \0 \0 \0 \0

Now let's assume I want to add these values -61 => \n => 35 => \n, starting at the beginning.  It should look like this:
0x100100cc     0x100100d0     0x100100d4     0x100100d8
- \n  5  3            \0 \n 1  6         \0 \0 \0 \0             \0 \0 \0 \0   <--- Blocks should be read in order, but backwards
  3  5  \n -      6  1 \n \0    ...

On the other hand, this is what YOUR output looks like:
0x100100cc     0x100100d0     0x100100d4     0x100100d8    0x100100dc
3 \0 \0 \0             5 \0 \0 \0          - \0 \0 \0            6 \0 \0 \0          1 \0 \0 \0
In other words, its writing the data backwards and in the wrong position.  (Actually its a bit more complicated than that because for each new number, you're writing over previously written data, but this should illustrate the point)

So, the conversion to ascii works fine, but adding the data to the buffer is incorrect.

Able to handle 50 lines? 4/4
Handles pos., neg. and higher magnitude integers? 5/5
Calls a sort method? 3/3
Sorts in ascending order? 5/5
Prints sorted integers to file? 1/4

Included 2 output files? 1/2
(Missing 1)

Included 2 input files? 1/2
(Missing 1)

PART 3:
Total: 0/45

No Submission?

TOTAL SCORE: 50/100
